src/service/viewer-impl.js

```diff
@@ -276,10 +276,17 @@ export class Viewer {
      */
     this.messagingReadyPromise_ = timer.timeoutPromise(
         20000,
-        new Promise(resolve => {
+        new Promise((resolve, reject) => {
           /** @private @const {function(!Viewer)} */
           this.messagingReadyResolver_ = resolve;
+
+          if (!this.isEmbedded_) {
+            reject(new Error('Page is not embedded.'));
```

Comment:
Throwing, then catching in the catch handler is the same thing with more code. :wink: